source/analyze-code.txt

```diff
@@ -0,0 +1,224 @@
+.. _visual-studio-extension-builders:
+
+=================
+Analyze Your Code
+=================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
```

Comment:
We probably should put some kind of disclaimer, that the generated MQL is "best effort" only, and the actual MQL generated at runtime can be different,  some known issues:

1. Code variables/constants appear in the generated MQL with their original code names, while the actual MQL will contain the actual value. For example:  For `Builders<Movie>.Filter.Gte(m => m.Score, minScore)` the analyzer generates `{ "Score": { "$gte": minScore } }` while in runtime it would be something like `{ "Score": { "$gte": 5.6 } }`.
2. Custom serialization settings will not be reflected by the current version or the analyzer.
3. Custom linq settings will not be reflected by the current version or the analyzer.
4. Not all valid expressions are successfully analyzed. For example: ```var names = new string[] { "Alice", "Bob" };
 _ = GetMongoQueryable<Person>().Where(p => p.Name == names[2]); ``` Will not generate any message, but we should emphasize that lack for translation does not mean that that is not a valid expression.
5. Currently we don't know of any, but in beta version there might be false positives and false negatives for unsupported expressions. Not sure what's the best way to mention that.

Some of those points appear in [Scope Doc.](https://docs.google.com/document/d/1ypNu93Ii3LleNLLOy6WLZwTRiD04zRLjPlN_tbQZEq0/edit#heading=h.b1os3ai9s8t3)

@mmarcon maybe we can just vaguely mention that this is beta/mvp and not dive into much details? Not sure what's the preferable approach here.

Reply:
Responding in order:

1. I've added an admonition mentioning variable names in output in the builders and LINQ sections
2. I think this can be added to the FAQ, under a heading "Does the MongoDB Analyzer use Custom Serialization Settings?" If you think this is appropriate I'll update the [JIRA ticket for the FAQ](https://jira.mongodb.org/browse/DOCSP-19710).
3. I think this can be added to the FAQ under a heading "Does the MongoDB Analyzer use Custom LINQ settings?" If you think this is appropriate I'll update the [JIRA ticket for the FAQ](https://jira.mongodb.org/browse/DOCSP-19710). 
4. Is there a way to classify the set of expressions that are not successfully analyzed? If this is done, I can make a "Warning" admonition informing readers to watch out for a known issue. If there is not a way to classify these expressions, it may be best to group this with the "False Positives" statment in point 5.
5. I'll reach out to the rest of the Developer Education team and see if there are any existing precedents for "Beta Release" docs. TO CONFIRM: Do we plan to market the MongoDB Analyzer as a "Beta" when its published on January 10th?